File: surf-cloud-api/src/main/java/dev/slne/surf/data/api/meta/SurfJpaRepository.java
package dev.slne.surf.data.api.meta;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;
import org.springframework.core.annotation.AliasFor;
import org.springframework.stereotype.Repository;

@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
@Repository
public @interface SurfJpaRepository {

  @AliasFor(annotation = Repository.class)
  String value() default "";
}

File: surf-cloud-api/surf-cloud-api-common/src/main/kotlin/dev/slne/surf/cloud/api/common/meta/SurfJpaRepository.kt
package dev.slne.surf.cloud.api.common.meta

import org.springframework.core.annotation.AliasFor
import org.springframework.stereotype.Repository

@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
@Repository
annotation class SurfJpaRepository(@get:AliasFor(annotation = Repository::class) val value: String = "")
